content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -173,22 +175,34 @@ class CreateAndIssueMarsVoucherInitiator @JsonConstructor constructor(private va
 2. Add the `call` method with the return type `SignedTransactionDigest`.
 3. Parse your parameters using the `mapOfParams` value. Since the parameters the flow will use come in JSON format, the JSON object must be parsed to extract the parameters so the flow can run.
-4. Add exceptions for when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
-5. Insert a method for finding the notary.
+4. Add appropriate error handling - exceptions must be thrown when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
+5. Insert the following method for finding the notary: `notaryLookup.getNotary(CordaX500Name.parse("O=notary, L=London, C=GB"))!!`.
 6. Build the output `MarsVoucher` state with a `UniqueIdentifier`.
-7. Build the transaction using a `txCommand` and the `transactionBuilderFactory` service you added when injecting services.
-8. Verify that the transaction is valid using the `txBuilder.verify` method.
-9. Sign the transaction using the `txBuilder.sign` method.
-10. Send the state to the counterparty and receive it back with their signature. Use the `flowMessaging` service to send the state to the `recipientParty`. Use the `flowEngine` service with the subflow `CollectSignaturesFlow` to get the signature of the counterparty.
-11. Notarize and record the transaction in both parties' vaults using `FinalityFlow` with a `fullySignedTx`.
-12. Return a `SignedTransactionDigest`. Use the `jsonMarshallingService` to parse the transaction's output. Include the transaction's output states, ID, and signatures.
+
+##### Build the transaction
+
+Your transaction is a proposal to the ledger that the counterparty must agree to. You must ensure that the proposal adheres the rules of the smart contract and sign it before sending it to the counterparty. The proposal here is that you'll create a voucher for a trip to Mars and then issue that voucher to a designated party - Peter.
+
+1. Build the transaction using a `txCommand` and the `transactionBuilderFactory` service you added when injecting services.
+2. Verify that the transaction is valid using the `txBuilder.verify` method.
+3. Sign the transaction using the `txBuilder.sign` method.
+
+##### Interact with the counterparty
+
+This is where you send the proposal to Peter. When consensus that the transaction's elements are unique and understood is achieved, the transaction is notarized in both parties' vaults.
+
+1. Send the state to the counterparty and receive it back with their signature. Use the `flowMessaging` service to send the state to the `recipientParty`. Use the `flowEngine` service with the subflow `CollectSignaturesFlow` to get the signature of the counterparty.
+2. Get the transaction notarized and recorded in both parties' vaults using `FinalityFlow` with a `fullySignedTx`.
+3. Return a `SignedTransactionDigest`. Use the `jsonMarshallingService` to parse the transaction's output. Include the transaction's output states, ID, and signatures.
```

Comment:
mention this bit is optional and only relevent to RPC invokeed flows as a way to return data to some interface

Reply:
updated